pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
Re. https://github.com/kubernetes/kubernetes/pull/13007/files#r38456017: Correct. I was suggesting a blanket policy. I do not think we need to nor should distinguish default values from other types of updates.

Example of user overriding automated process: Setting resources. User may initially rely on automatically set resources and then may need to override them later. Or just changing a value from the default to something else.

We're not trying to solve the problem of a continuous automatic process repeatedly stomping the user's config. That's a different issue.